src/backend/common/Pipeline.cpp

```diff
@@ -27,7 +28,7 @@ namespace backend {
     PipelineBase::PipelineBase(PipelineBuilder* builder)
         : device(builder->device), stageMask(builder->stageMask), layout(std::move(builder->layout)),
           renderPass(std::move(builder->renderPass)), subpass(builder->subpass),
-          inputState(std::move(builder->inputState)) {
+          inputState(std::move(builder->inputState)), depthStencilState(std::move(builder->depthStencilState)) {
```

Comment:
Can you add validation in this function that if IsCompute then we don't have non-compute state?